pkg/metrics/stats_reporter.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+	"strconv"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/metric"
+	"go.opentelemetry.io/otel/label"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/secrets-store-csi-driver/apis/v1alpha1"
+)
+
+var (
+	secretProviderClassKey    = "secretproviderclass"
+	namespaceKey              = "namespace"
+	secretObjectName          = "secret_name"
+	secretObjectTypeKeyPrefix = "secret_type_"
+	providerKey               = "provider"
+	labelKeyPrefix            = "label_"
+
```

Comment:
I think we should also add the os type to the metrics

https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/dcf491d1c4b3fee8cec6320d2d9224cfae57cc2c/pkg/secrets-store/stats_reporter.go#L28

Reply:
added